crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -161,12 +161,20 @@ impl Rule for ConsistentFunctionScoping {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
         let (function_declaration_symbol_id, function_body, reporter_span) = match node.kind() {
             AstKind::Function(function) => {
-                if let Some(AstKind::AssignmentExpression(_)) = ctx.nodes().parent_kind(node.id()) {
+                if function.is_typescript_syntax() {
                     return;
                 }
 
-                if function.is_typescript_syntax() {
```

Comment:
Why remove this logic?

Reply:
lets add a test case for this? does linter tests allow .d.ts?

Reply:
> Why remove this logic?

I did not. It's in here:

https://github.com/Arian94/oxc/blob/3af5abf774ac299a83917d25b361627b28277912/crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs#L164